apps/editing-toolkit/editing-toolkit-plugin/editor-header-buttons/src/index.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External Dependencies
+ */
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import domReady from '@wordpress/dom-ready';
+import { PostPreviewButton, PostPublishButton, PostSavedState } from '@wordpress/editor';
+import { Button, Dropdown } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+/**
+ * Internal Dependencies
+ */
+
+import './styles.scss';
+
+const PostEditorButtonMenu: React.FunctionComponent = () => {
+	const editPostStore = 'core/edit-post';
+
+	const { hasActiveMetaboxes, isPublishSidebarOpened, isSaving } = useSelect(
+		( select ) => ( {
+			hasActiveMetaboxes: select( editPostStore ).hasMetaBoxes(),
+			isPublishSidebarOpened: select( editPostStore ).isPublishSidebarOpened(),
+			isSaving: select( editPostStore ).isSavingMetaBoxes(),
+			showIconLabels: select( editPostStore ).isFeatureActive( 'showIconLabels' ),
+			hasReducedUI: select( editPostStore ).isFeatureActive( 'reducedUI' ),
+			isEditingTemplate: select( editPostStore ).isEditingTemplate(),
+		} ),
+		[]
+	);
+
+	// These buttons may require props that exist for their usages in Gutenberg here; gutenberg-repo/packages/edit-post/src/components/header/index.js
+	return (
+		<div className="dropdown-container">
+			<Dropdown
+				className="my-container-class-name"
+				contentClassName="my-popover-content-classname"
+				position="bottom center"
+				renderToggle={ ( { isOpen, onToggle } ) => (
+					<Button isPrimary onClick={ onToggle } aria-expanded={ isOpen }>
+						Update
+					</Button>
+				) }
+				renderContent={ () => (
+					<div className="dropdown-content">
```

Comment:
These components may require additional props which we will also need to look into. You can find their usages [here in Gutenberg](https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/header/index.js#L67-L94)